pilot/pkg/networking/core/v1alpha3/tracing.go

```diff
@@ -314,7 +312,26 @@ func otelConfig(serviceName, hostname, cluster string, otelProvider *meshconfig.
 		}
 	}
 
-	oc.ServiceName = serviceName
+	// Add configured resource detectors
+	if otelProvider.ResourceDetectors != nil {
+		res := []*core.TypedExtensionConfig{}
+		rd := otelProvider.ResourceDetectors
+
+		if rd.Environment != nil {
+			res = append(res, &core.TypedExtensionConfig{
+				Name:        "envoy.tracers.opentelemetry.resource_detectors.environment",
+				TypedConfig: protoconv.MessageToAny(&resourcedetectors.EnvironmentResourceDetectorConfig{}),
+			})
+		}
+		if rd.Dynatrace != nil {
```

Comment:
are multiple supported?

Reply:
No, it doesn't make sense to support multiple of the same as they would simply override each other (the data it produces)

Reply:
but we are appending here, so someone could set dynatrace and environment. Your comment above sounds like that is not expected?

Reply:
it's expected, but should avoid multiple detectors in same type(either dynatrace or environment).

Reply:
> but we are appending here, so someone could set dynatrace and environment. Your comment above sounds like that is not expected?

Ah right, sorry. Yes it is totally possible to have multiple resource detectors configured - AWS, Environment, Dynatrace. All of them detect different attributes and merge all into a final Resource. 

What I meant is that it doesn't make sense to configure the same detector twice, as they would just override each other.

Reply:
ACK, thanks!